instrumentation/opentelemetry-instrumentation-aws-lambda/src/opentelemetry/instrumentation/aws_lambda/coralogix/span/__init__.py

```diff
@@ -0,0 +1,79 @@
+"""
+This module provides functionality for creating OpenTelemetry spans for AWS Lambda functions.
+
+It contains the base CoralogixSpan class that defines the interface for creating spans for different
+AWS Lambda triggers, as well as concrete implementations for various AWS services like:
+
+"""
+from typing import Dict, Tuple, Any
+from dataclasses import dataclass, field
+from abc import ABC, abstractmethod
+
+from opentelemetry.semconv.trace import SpanAttributes
+from opentelemetry.trace import Tracer, Span, SpanKind, Context
+
+from opentelemetry.instrumentation.aws_lambda.coralogix import attributes as cx_attributes
+
+@dataclass
+class CoralogixSpan(ABC):
```

Comment:
Would it be possible to somehow just indicate how to find the response/request body from the event (if it has any) and the span attribute name to store it so that we don't need to repeat this kind of snippet everywhere:

```python
        if first_record.get("Sns") and first_record["Sns"].get("Message"):  # type: ignore[unknown-get]
            body = limit_string_size(first_record["Sns"].get("Message"))  # type: ignore[unknown-get]
            sns_trigger_span.set_attribute(
                cx_attributes.RPC_REQUEST_BODY,
                body,
            )
````

Asking because I think it's kinda error prone. It's relatively easy to forget to apply the limit, for instance.

Reply:
Yes. I didn't apply this because there is something that is different in the others events.. but I can create it and override.

Reply:
It can be done later, imho, but I think we should do it to prevent mistakes.